File: hw/toapanta/u3/hw35CrudStrategy/CrudStrategyJS/models/strategies/DeleteStrategy.js
export class DeleteStrategy {
    constructor(storageStrategy) {
        this.storageStrategy = storageStrategy;
    }

    async execute(customerId) {
        const customer = await this.storageStrategy.readById(customerId);
        if (!customer) {
            console.log(`Error: ID ${customerId} not found in ${this.storageStrategy.getFormatName()}`);
            return false;
        }
        
        return await this.storageStrategy.delete(customerId);
    }

    setStorageStrategy(storageStrategy) {
        this.storageStrategy = storageStrategy;
    }

    getStorageFormatName() {
        return this.storageStrategy.getFormatName();
    }
}

File: hw/pastor/u3/hw35CrudStrategy/CrudStrategyPY/model/strategy/delete_strategy.py
from model.strategy.crud_strategy import CrudStrategy
from model.customer import Customer

class DeleteStrategy:
    def __init__(self, storage_strategy: CrudStrategy):
        self.storage_strategy = storage_strategy
    
    def execute(self, customer_id: int) -> bool:
        customer = self.storage_strategy.read_by_id(customer_id)
        if not customer:
            print(f"Error: ID {customer_id} not found in {self.storage_strategy.get_format_name()}")
            return False
        
        return self.storage_strategy.delete(customer_id)
    
    def set_storage_strategy(self, storage_strategy: CrudStrategy):
        self.storage_strategy = storage_strategy
    
    def get_storage_format_name(self) -> str:
        return self.storage_strategy.get_format_name()